### PR TITLE
DAS-282 Improved charting (JS)

### PR DESCRIPTION
**Pull request by AndrewSazonov**
The following has been implemented:

Python backend: 
* Unified class for plotting 1d data. It includes creation of data container and data object accessible in both Bokeh and QtCharts QML frontends.

QML frontend: 
* easyAppGui: Unified plotting base for Bokeh and QtCharts (JS-based): https://github.com/easyScience/easyAppGui/pull/5
* easyDiffractionApp: Experiment and Analysis charts with measured, calculated and difference curves, as well as Bragg peaks and Background curve

### CODE REVIEW

**Comment by wardsimon** [comment]
Re-opened for review

File: easyDiffractionApp/Gui/Components/AnalysisDataChartBokeh.qml [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 
-    xAxisTitle: qsTr("2theta (deg)")
-    yAxisTitle: qsTr("Intensity")
+    plotRanges: ExGlobals.Constants.proxy.plotting1d.analysisPlotRangesObj
 
-    experimentLineColor: EaStyle.Colors.chartForegrounds[0]
-    calculatedLineColor: EaStyle.Colors.chartForegrounds[1]
+    xAxisTitle: qsTr("2theta (deg)")
+    yMainAxisTitle: qsTr("Intensity")
+    yDifferenceAxisTitle: qsTr("Difference")
```
Comment (arm61) [inline easyDiffractionApp/Gui/Components/AnalysisDataChartBokeh.qml:19]: Difference or residual?

File: easyDiffractionApp/main.py [hunk 2]
```diff
@@ -26,79 +18,41 @@ def isTestMode():
...
-            return
-
 def main():
-    # Settings
-    QCoreApplication.setAttribute(Qt.AA_EnableHighDpiScaling)  # DOESN'T WORK, USE SCRIPT INSTEAD
-    QCoreApplication.setAttribute(Qt.AA_UseDesktopOpenGL, True)
+    # Arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-l', '--logtofile', action='store_true',
+                        help='enable logging in the file easyDiffraction.log in the system directory tmp instead of the terminal')
```
Comment (arm61) [inline easyDiffractionApp/main.py:35]: `easyDiffraction.log` -> `f{appname}.log`?

**Comment by arm61** [comment]
Plotting appears to work nicely in testing of this branch.

**Comment by rozyczko** [comment]
closed as no longer relevant.

**Pull request closed without merging**